.github/workflows/build.yml

```diff
@@ -117,9 +117,12 @@ jobs:
       uses: codecov/codecov-action@v4.1.0
     - name: Install jars
       run: ./mvnw install -q -B -DskipTests
+    - name: Integration tests with testcontainers using spring boot version ${{ matrix.spring-boot-version }}
```

Comment:
Why this needs to be separate?

Reply:
@artursouza good question, this helps to isolate tests running with testcontainers from tests running with the local `daprd` binary. Running both concurrently generates issues that are hard to reproduce locally. I think it is sensitive to separate them until we can align both approaches better.

Reply:
As @salaboy mentioned, when running everything together we have seen pretty weird behavior where GRPC channel is not properly closed and things like that, which lead to errors that are hard to reproduce in isolation. So the only way to overcome this problem was to have two groups of test:
- Testcontainers based - these are ITs annotated with `@Tag("testcontainers")` and are relying on Testcontainers library to start, stop and cleanup any started containers.
- Other ITs - these are ITs that are using `daprd` process with Docker Compose etc

Reply:
This is a problem. It means the tests are leaving something running behind and influencing with other tests. If we merge this, it will be a technical debt. I don't mind having the Tag for people to run them separately locally, but they should run together. It is also more cognitive burden on contributions to figure out that cannot run the ITs together.

Reply:
If this cannot be solved in this PR, I will still merge but I will create an issue for someone to fix this tech debt.